### PR TITLE
Add ALACRITTY_RENDERER environment variable

### PR DESCRIPTION
**Pull request by kchibisov**
This should help trouble shooting the renderer being created and different renderer options to determine when something like dual-source rendering isn't working.

### CODE REVIEW

**Review by chrisduerr — COMMENTED** [review]
Why did you decide to go with an environment variable rather than a configuration option?

As far as I remember, this would be the first thing we expose as an environment variable in Alacritty directly.

**Comment by kchibisov** [comment]
Hm, I'm just used that most libs control the renderer to use that way. Could add a debug option for that. It's not like we directly expose it. It'll be used in a context with 'could you test renderer XXX'.

Do you want a `debug` option for that?

**Comment by kchibisov** [comment]
I've updated to use the config option.

File: alacritty.yml [hunk 1]
```diff
@@ -888,6 +888,12 @@
   #   - Trace
   #log_level: Warn
 
+  # Renderer to force
```
Comment (chrisduerr) [inline alacritty.yml:891]:
```suggestion
  # Renderer override.
```

File: alacritty/src/config/debug.rs [hunk 2]
```diff
@@ -18,6 +20,9 @@ pub struct Debug {
     /// Highlight damage information produced by alacritty.
     pub highlight_damage: bool,
 
+    /// The renderer alacritty should be using
```
Comment (chrisduerr) [inline alacritty/src/config/debug.rs:23]:
```suggestion
    /// The renderer alacritty should be using.
```

File: alacritty/src/config/debug.rs [hunk 3]
```diff
@@ -32,6 +37,23 @@ impl Default for Debug {
...
+/// The renderer configuration options.
+#[derive(Deserialize, Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub enum RendererPreference {
+    /// OpenGL 3.3 renderer.
+    #[serde(rename = "glsl3")]
+    Glsl3,
+
+    /// GLES 2 renderer, but can use extensions like dual source blending.
+    #[serde(rename = "gles2")]
+    Gles2,
```
Comment (chrisduerr) [inline alacritty/src/config/debug.rs:54]:
```suggestion
    /// GLES 2 renderer, with optional extensions like dual source blending.
    #[serde(rename = "gles2")]
    Gles2,
```

**Review by chrisduerr — APPROVED** [review]

**Review by chrisduerr — APPROVED** [review]

**Pull request merged**